### ISSUE TITLE
Array declaration is parsed wrong

### ISSUE THREAD

**Issue by Miglecz** [javaparser/javaparser#416]
Compiler gives an error on the following (stating the the method is already defined):

```
public void m1(boolean[] boolArray) {}
public void m1(boolean boolArray[]) {}
```

But the parser creates different objects out of it.

`boolean boolArray[]` is not recognised the same way as the other one. It is treated as a simple `PrimitiveType`.

**Comment by matozoid** [comment]
Okay, detecting duplicate methods is indeed the task of the compiler. Syntactically it is correct, so JavaParser parses it.

The same can be said about those two not being the same.

``` java
        CompilationUnit parse = JavaParser.parse(new StringReader("public class Foo {\n" +
                "    boolean[] boolArray;\n" +
                "    boolean boolArray[];\n" +
                "}\n"));

        List<FieldDeclaration> fields = parse.getClassByName("Foo").getFields();

        assertEquals(1, ((ReferenceType)fields.get(0).getType()).getArrayCount());
        assertEquals(1, fields.get(1).getVariables().get(0).getId().getArrayCount());
```

As you can see there was no information lost, it is just stored in another place, attached to either the type or the id (name) of the field.

Is that a good design? I'm not sure.

**Comment by matozoid** [comment]
No more comments?

**Comment by matozoid** [comment]
I guess it's fine then.

**Comment by matozoid** [comment]
This is actually being addressed in #237 now.
